gridscale/resource_gridscale_loadbalancer.go

```diff
@@ -189,23 +182,25 @@ func resourceGridscaleLoadBalancerUpdate(d *schema.ResourceData, meta interface{
 		Algorithm:           d.Get("algorithm").(string),
 		Status:              d.Get("status").(string),
 		RedirectHTTPToHTTPS: d.Get("redirect_http_to_https").(bool),
-		ListenIPv4Uuid:      d.Get("listen_ipv4_uuid").(string),
-		ListenIPv6Uuid:      d.Get("listen_ipv6_uuid").(string),
-		Labels:              d.Get("labels").(*schema.Set).List(),
+		ListenIPv4UUID:      d.Get("listen_ipv4_uuid").(string),
+		ListenIPv6UUID:      d.Get("listen_ipv6_uuid").(string),
+		Labels:              convSOStrings(d.Get("labels").(*schema.Set).List()),
 	}
-
 	if backendServers, ok := d.GetOk("backend_server"); ok {
 		requestBody.BackendServers = expandLoadbalancerBackendServers(backendServers)
 	}
 	if forwardingRules, ok := d.GetOk("forwarding_rule"); ok {
 		requestBody.ForwardingRules = expandLoadbalancerForwardingRules(forwardingRules)
 	}
 	err := client.UpdateLoadBalancer(d.Id(), requestBody)
-
 	if err != nil {
 		return fmt.Errorf(
 			"Error waiting for loadbalancer (%s) to be updated: %s", requestBody.Name, err)
 	}
+	err = pauseWhenProvisoning(client, loadbalancerService, d.Id())
```

Comment:
This can be replaced with ```	return resource.Retry(d.Timeout(schema.TimeoutUpdate), func() *resource.RetryError {
		res, err := client.GetLoadBalancer(d.Id())

		if err != nil {
			return resource.NonRetryableError(fmt.Errorf("Error getting loadbalancer3: %s", err))
		}

		if res.Properties.Status == provivisoningStatus {
			return resource.RetryableError(fmt.Errorf("Expected loadbalancer to be updated but was in state %s", res.Properties.Status))
		}

		return resource.NonRetryableError(resourceGridscaleLoadBalancerRead(d, meta))
	})```